ui/types/utils.d.ts

```diff
@@ -103,11 +103,15 @@ export function setCssVar(
   element?: Element
 ): void;
 
-export class EventBus {
-  on(event: string, callback: Function, ctx?: any): this;
-  once(event: string, callback: Function, ctx?: any): this;
-  emit(event: string, ...args: any[]): this;
-  off(event: string, callback?: Function): this;
+interface Callbacks {
+  [key: string]: (...args: any[]) => void;
+}
+
+export class EventBus<T extends Callbacks> {
+  on<K extends keyof T>(event: K, callback: T[K], ctx?: any): this;
+  once<K extends keyof T>(event: K, callback: T[K], ctx?: any): this;
```

Comment:
The type of `ctx` and whether it should be required or not could be derived from `T[K]` using `ThisParameterType` to bring in more features.

Reply:
@yusufkandemir Can you PR this, pls?

Reply:
I can fix this later too

Reply:
Will make a new Quasar release tomorrow. Do you think you can PR it by then, pretty pls?

Reply:
Created a new PR #15765